### PR TITLE
ec2_asg_find.py will allow you to search for auto scale groups.

### PR DESCRIPTION
**Pull request by linuxdynasty**
Search AutoScaleGroups based on the name or prefix of the name.

### CODE REVIEW

**Comment by gregdek** [comment]
Thanks @linuxdynasty for this new module. When this module receives 'shipit' comments from two community members and any 'needs_revision' comments have been resolved, we will mark for inclusion.

**Comment by manicminer** [comment]
@linuxdynasty Awesome, exactly what I was looking for! I'd like to add additional search parameters, specifically enabling search by matching tags. I already have working code for this in another module that I'm busy with, although it uses boto3. Would you be ok with me updating this module to use boto3, and adding a `tags` parameter?

Some other comments to follow; needs_revision

File: cloud/amazon/ec2_asg_find.py [hunk 1]
```diff
@@ -0,0 +1,430 @@
+#!/usr/bin/env python
```
Comment (manicminer) [inline cloud/amazon/ec2_asg_find.py]: Shebang should be `#!/usr/bin/python`

File: cloud/amazon/ec2_asg_find.py [hunk 2]
```diff
@@ -0,0 +1,430 @@
...
+# You should have received a copy of the GNU General Public License
+# along with Ansible.  If not, see <http://www.gnu.org/licenses/>.
+
+DOCUMENTATION = '''
+---
+module: ec2_asg_find
+short_description: Find autoscale groups.
+description:
+    - Find autoscale groups by name and retrieve the properties of each one.
+version_added: "0.1"
```
Comment (manicminer) [inline cloud/amazon/ec2_asg_find.py]: Should match the next planned version of Ansible, perhaps 2.1?

**Comment by sivel** [comment]
Here are the results of automated validation:

```
============================================================================
ec2_asg_find.py
============================================================================
Traceback (most recent call last):
ParserError: while parsing a block mapping
  in "<string>", line 2, column 5:
        description: The Amazon ARN name ...
        ^
expected <block end>, but found ','
  in "<string>", line 5, column 159:
     ... Name/public-webapp-production-1",
                                         ^

ERROR: version_added should be 2.1. Currently 0.1
ERROR: RETURN is not valid YAML
ERROR: Interpreter line is not "#!/usr/bin/python"
```

**Comment by manicminer** [comment]
I've updated this proposed module locally with these additions/changes:
- Use boto3
- Accept a `tags` parameter to search auto scaling groups with matching tags
- Made `name` and `tags` optional, omitting them returns all groups in the current region
- Revised as per above comments

I would like to contribute these changes, should I open a new PR?

**Comment by linuxdynasty** [comment]
@tombamford I think if you fork my fork my branch and create a PR on it and add yourself as an author, since we are both collaborating on it. I will check it out and merge it in and see if this PR gets updated with your changes.

**Comment by manicminer** [comment]
@linuxdynasty Great, I've done just that; see https://github.com/linuxdynasty/ansible-modules-extras/pull/1. Thanks!

**Comment by linuxdynasty** [comment]
@tombamford great work and thanks for the well documented updates in the code and for fixing my typos ;).

**Comment by manicminer** [comment]
@linuxdynasty Awesome, thanks for merging. Contribution guidelines for Ansible require squashing of merge commits, would you be able to do that? Hopefully this can get reviewed soon.

**Comment by manicminer** [comment]
@linuxdynasty also, fixed broken indentation with https://github.com/linuxdynasty/ansible-modules-extras/pull/2

**Comment by linuxdynasty** [comment]
@tombamford I think I should create a new PR, so that I can rebase the commits and not force push them up. I think this is the only way to follow the guidelines.

**Comment by linuxdynasty** [comment]
I'm closing this PR for #1659

**Comment by tombamford** [comment]
@linuxdynasty I don't think force pushing your existing branch would cause any problems. Code comments will not be lost and it will preserve this discussion thread.

**Pull request closed without merging**